Source/WebCore/platform/graphics/PlatformColorSpace.h

```diff
@@ -39,6 +42,26 @@ namespace WebCore {
 using PlatformColorSpace = RetainPtr<CGColorSpaceRef>;
 using PlatformColorSpaceValue = CGColorSpaceRef;
 
+#elif USE(SKIA)
+
+class PlatformColorSpace {
+public:
+    explicit PlatformColorSpace(Vector<uint8_t> serializedColorSpace) { deserialize(serializedColorSpace); }
```

Comment:
So, if I understand this correctly now, we need this wrapper class instead of using PlatformColorSpace as sk_sp<SkColorSpace> directly only for the serialization, right? I don't know much about the new serializers, but would it be possible to add a CoreIPC wrapper of SkColorSpace to serialize/deserialize it and use PlatformColorSpace as sk_sp<SkColorSpace> like CG does for CGColorSpaceRef?

Reply:
Correct - if it wasn't for serialization we could just use `sk_sp<SkColorSpace>` as `PlatformColorSpace` and `PlatformColorSpaceValue`.
As for `CoreIPC wrapper` I'm not sure what you mean, but if it's about providing custom serializer, we'd have to figure out how to avoid generating default ones in `WebCoreArgumentCoders.serialization.in` as I've found it not trivial lat time I've checked.